### DISCUSSION TITLE
get video snapshot

### DISCUSSION THREAD

**Discussion by ymmmvp36** [alexmercerind/dart_vlc#118] [Q&A] [answered]
hi,is there someway to get the snapshot of the video?

**Comment by alexmercerind** [comment]
There is `libvlc_video_take_snapshot`, but not exposed to Dart yet. You may open issue for this, so that we can track it.
I cannot give any expected date for this to land however, we're still working to make everything better on Linux & macOS specifically.
Thanks.

**Reply by ymmmvp36** [reply]
thank you.in the version 0.0.8. i put video under the widget RepaintBoundary. when the playbackController tells me the video is playing. i will use the RepaintBoundary to get the image. i found in the version 0.1.4 this method does not work. the RepaintBoundary cannot get the texture content.do u have some idea about this?

> There is `libvlc_video_take_snapshot`, but not exposed to Dart yet. You may open issue for this, so that we can track it.
> I cannot give any expected date for this to land however, we're still working to make everything better on Linux & macOS specifically.
> Thanks.

**Reply by alexmercerind** [reply]
@ymmmvp36 in earlier versions (when texture support for Windows was not available), I used to decode raw image into Flutter's `Image` (`decodeImageFromPixels`) widget, but now the video playback uses `Texture` widget. This is much much more performant than before however. Possibly it is the reason.

**Comment by alexmercerind** [comment, marked as answer]
This is now available on master

https://github.com/alexmercerind/dart_vlc/blob/70901d93f5b877c14b58d6d577d5000701e1ece5/ffi/lib/src/player.dart#L273-L276

**Reply by ymmmvp36** [reply]
great!
